components/script/script_thread.rs

```diff
@@ -1829,29 +1810,15 @@ impl ScriptThread {
                     self.handle_event(id, event)
                 },
                 FromScript(MainThreadScriptMsg::Common(CommonScriptMsg::Task(
-                    category,
-                    task,
-                    pipeline_id,
+                    _,
+                    _,
+                    _,
                     TaskSourceName::Rendering,
                 ))) => {
-                    if rendering_update_already_prioritized {
-                        // If we've already updated the rendering,
-                        // run this task along with the other non-prioritized ones.
-                        sequential.push(FromScript(MainThreadScriptMsg::Common(
-                            CommonScriptMsg::Task(
-                                category,
-                                task,
-                                pipeline_id,
-                                TaskSourceName::Rendering,
-                            ),
-                        )));
-                    } else {
-                        // Run the "update the rendering" task.
-                        task.run_box();
-                        // Always perform a microtrask checkpoint after running a task.
-                        self.perform_a_microtask_checkpoint(can_gc);
-                        rendering_update_already_prioritized = true;
-                    }
+                    // Instead of running this task now or while handling other messages, `update_the_rendering()`
```

Comment:
It's not clear to me what the meaning of "atomic", and "consistent" are, and also not all messages here are from IPC(for example this message is not), and I would rather keep the comment minimal and descriptive(as opposed to prescriptive but with unclear intent). We can eventually consolidate all docs into a section in the book. 

I propose: "Instead of interleaving any number of update the rendering tasks with other message handling, we run those steps only once at the end of each call of `handle_msgs`."

Reply:
Good point. I've used your proposed comment with some minor changes.